### PR TITLE
switch rbac to external

### PR DESCRIPTION
**Pull request by deads2k**
The overall trajectory of the project is towards external types.  Having all helpers agree on the version they operate on makes life much easier.  We've already written one RBAC controller (role aggregation) and more may follow.  `v1` has been around for a while now and we know that any future changes have to reliably roundtrip through it.  This pull switches all the core helpers over to use the external types.

@kubernetes/sig-auth-pr-reviews 

```release-note
`kubectl auth reconcile` only works with rbac.v1
```

### CODE REVIEW

File: test/integration/framework/master_utils.go [hunk 1]
```diff
@@ -199,17 +199,20 @@ func startMasterOrDie(masterConfig *master.Config, incomingServer *httptest.Serv
 		closeFn()
 		glog.Fatal(err)
 	}
+	var lastHealthContent []byte
```
Comment (ericchiang) [inline test/integration/framework/master_utils.go]: Is this supposed to be part of the PR?

**Review by ericchiang — COMMENTED** [review]
Why the changes to `pkg/registry/rbac`? Was something else importing it?

**Comment by deads2k** [comment]
I think this is ready now.

File: pkg/registry/rbac/clusterrolebinding/policybased/storage.go [hunk 2]
```diff
@@ -59,7 +60,7 @@ func (s *Storage) Create(ctx context.Context, obj runtime.Object, createValidati
 		return s.StandardStorage.Create(ctx, obj, createValidation, includeUninitialized)
 	}
 
-	rules, err := s.ruleResolver.GetRoleReferenceRules(clusterRoleBinding.RoleRef, metav1.NamespaceNone)
+	rules, err := s.ruleResolver.GetRoleReferenceRules(rbacv1helpers.ConvertOrDieRoleRef(clusterRoleBinding.RoleRef), metav1.NamespaceNone)
```
Comment (liggitt) [inline pkg/registry/rbac/clusterrolebinding/policybased/storage.go]: `___OrDie` inside a rest handler isn't something I'd expect

File: pkg/kubectl/cmd/auth/reconcile.go [hunk 3]
```diff
@@ -161,13 +166,13 @@ func (o *ReconcileOptions) RunReconcile() error {
 		}
 
 		switch t := info.Object.(type) {
-		case *rbac.Role:
+		case *rbacv1.Role:
```
Comment (liggitt) [inline pkg/kubectl/cmd/auth/reconcile.go]: does this mean `kubectl auth reconcile` only works on rbac/v1 resources? that's notable

Comment (deads2k) [reply pkg/kubectl/cmd/auth/reconcile.go]: > does this mean kubectl auth reconcile only works on rbac/v1 resources? that's notable

Yes.  It is consistent with the idea of kubectl moving to external versions and skewing only by a release.

**Comment by deads2k** [comment]
@sttts any idea what I've done with this generated deep copy?

File: pkg/registry/rbac/clusterrolebinding/policybased/storage.go (hunk 2 above)
Comment (deads2k) [reply pkg/registry/rbac/clusterrolebinding/policybased/storage.go]: > ___OrDie inside a rest handler isn't something I'd expect

I'm willing to change it, but it is a little difficult to come up with a way it won't round trip.

**Comment by deads2k** [comment]
Added release note, removed use of OrDie.

Might have to chase the generation ghost

**Comment by liggitt** [comment]
switch to external looks good
/lgtm

File: pkg/apis/rbac/v1/zz_generated.deepcopy.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by conversion-gen. DO NOT EDIT.
+
+package v1
```
Comment (sttts) [inline pkg/apis/rbac/v1/zz_generated.deepcopy.go]: this is pretty empty

Comment (sttts) [reply pkg/apis/rbac/v1/zz_generated.deepcopy.go]: I guess you want a doc.go with `+// +k8s:deepcopy-gen=package`.

**Comment by deads2k** [comment]
Alright, think I sorted out my generation problem, retagged.

**Pull request merged**